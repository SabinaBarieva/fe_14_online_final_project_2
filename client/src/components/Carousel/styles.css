.swiper {
  padding-bottom: 30px;
}

.swiper-slide {
  text-align: center;
}

.swiper-pagination {
  display: flex;
  flex-wrap: nowrap;
  justify-content: center;
}
.swiper-pagination-fraction {
  bottom: 0;
}
.swiper-pagination-bullet {
  width: 25px;
  height: 5px;
  border-radius: 5px;
}
.swiper-pagination-bullet-active {
  background: rgb(46, 45, 45);
}

.swiper-button-prev,
.swiper-button-next {
  color: rgb(202, 199, 199);
}
.swiper-button-prev:hover,
.swiper-button-next:hover {
  color: rgb(46, 45, 45);
}

/*# sourceMappingURL=styles.css.map */
