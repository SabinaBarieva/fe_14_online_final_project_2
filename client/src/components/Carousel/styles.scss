.swiper {
  @media screen and (max-width: 768px) {
    width: 100%;
  }

  width: 97%;
  padding-bottom: 30px;
}

.swiper-slide {
  text-align: center;
}

.swiper-pagination {
  display: flex;
  flex-wrap: nowrap;
  justify-content: center;

  &-fraction {
    bottom: 0;
  }

  &-bullet {
    width: 25px;
    height: 5px;
    border-radius: 5px;

    &-active {
      background: rgb(46, 45, 45);
    }
  }
}

.swiper-button-prev,
.swiper-button-next {
  color: rgb(202, 199, 199);

  &:hover {
    color: rgb(46, 45, 45);
  }
}
