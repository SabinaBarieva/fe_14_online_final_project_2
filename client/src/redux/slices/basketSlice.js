import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  priceAll: 0,
  itemsBasket: [],
  modal: false,
  modalText: '',
};
const basketSlice = createSlice({
  name: 'basket',
  initialState,
  reducers: {
    addToBasket: (state, action) => {
      const seachItem = state.itemsBasket.find(
        (item) => item.itemNo === action.payload.itemNo
      );
      if (seachItem) {
        if (seachItem.count !== seachItem.quantity) {
          seachItem.count += 1;
        } else {
          state.modal = true;
          state.modalText = `Sorry, there are only ${seachItem.quantity} units of this product in stock`;
        }
      } else {
        state.itemsBasket.push({
          ...action.payload,
          count: 1,
        });
      }
      state.priceAll = state.itemsBasket.reduce(
        (sum, item) => item.currentPrice * item.count + sum,
        0
      );
    },
    minusItem: (state, action) => {
      const findItem = state.itemsBasket.find(
        (obj) => obj.itemNo === action.payload.itemNo
      );
      if (findItem) {
        findItem.count -= 1;
      }
      state.priceAll = state.itemsBasket.reduce(
        (sum, item) => item.currentPrice * item.count + sum,
        0
      );
    },
    deleteBasket: (state, action) => {
      state.itemsBasket = state.itemsBasket.filter(
        (item) => item.itemNo !== action.payload.itemNo
      );
      state.priceAll = state.itemsBasket.reduce(
        (sum, item) => item.currentPrice * item.count + sum,
        0
      );
    },
    clearBasket(state) {
      state.itemsBasket = [];
      state.priceAll = [];
    },
    // eslint-disable-next-line consistent-return
    addSeveraltoBasket: (state, action) => {
      const seachItem = state.itemsBasket.find(
        (item) => item.itemNo === action.payload.itemNo
      );
      if (seachItem) {
        if (
          action.payload.count + seachItem.count > seachItem.quantity &&
          seachItem.quantity - seachItem.count !== 0
        ) {
          state.modalText = `You can't add more than ${
            seachItem.quantity - seachItem.count
          }`;
          state.modal = true;
          return;
        }
        if (seachItem.count !== seachItem.quantity) {
          seachItem.count += action.payload.count;
        } else {
          state.modal = true;
          state.modalText = `Sorry, there are only ${seachItem.quantity} units of this product in stock`;
        }
      } else {
        state.itemsBasket.push({
          ...action.payload,
          count: action.payload.count,
        });
      }
      state.priceAll = state.itemsBasket.reduce(
        (sum, item) => item.currentPrice * item.count + sum,
        0
      );
    },
    closeModalBasket(state) {
      state.modal = false;
      state.modalText = '';
    },
  },
});
export const {
  addToBasket,
  deleteBasket,
  clearBasket,
  minusItem,
  addSeveraltoBasket,
  closeModalBasket,
} = basketSlice.actions;
export default basketSlice.reducer;
