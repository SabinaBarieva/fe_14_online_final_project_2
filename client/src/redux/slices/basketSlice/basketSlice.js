import { createSlice } from '@reduxjs/toolkit';
import { login } from '../loginSlice';
import mergeBasket from './mergeBasket';
import changeQuantityInBasket from './changeQuantity';
import deleteFromBasket from './deleteFromBasket';

const calcBasketPriceSum = (productsInBasket) =>
  productsInBasket.reduce(
    (sum, { product, cartQuantity }) =>
      product.currentPrice * cartQuantity + sum,
    0
  );

const initialState = {
  priceAll: 0,
  itemsBasket: [],
  modal: false,
  modalText: '',
};

const basketSlice = createSlice({
  name: 'basket',
  initialState,
  reducers: {
    closeModalBasket: (state) => {
      state.modal = false;
      state.modalText = '';
      state.modalAdd = false;
    },
    clearBasket: (state) => {
      state.itemsBasket = [];
    },
  },
  extraReducers: (builder) => {
    builder.addCase(login.fulfilled, (state, action) => {
      action.asyncDispatch(mergeBasket());
    });
    builder.addCase(mergeBasket.fulfilled, (state, { payload: cart }) => {
      state.itemsBasket = cart;
      state.priceAll = calcBasketPriceSum(state.itemsBasket);
    });
    builder.addCase(
      changeQuantityInBasket.fulfilled,
      (state, { payload: cart }) => {
        state.itemsBasket = cart;
        state.priceAll = calcBasketPriceSum(cart);
      }
    );
    builder.addCase(
      changeQuantityInBasket.rejected,
      (state, { payload: { error, type, message } }) => {
        if (type === 'quantityError') {
          state.modalText = message;
          state.modal = true;
        } else {
          console.log(error);
        }
      }
    );
    builder.addCase(deleteFromBasket.fulfilled, (state, { payload: cart }) => {
      state.itemsBasket = cart;
      state.priceAll = calcBasketPriceSum(cart);
    });
  },
});

export const {
  addToBasket,
  deleteBasket,
  clearBasket,
  minusItem,
  addSeveraltoBasket,
  closeModalBasket,
  updateBasket,
} = basketSlice.actions;

export default basketSlice.reducer;
