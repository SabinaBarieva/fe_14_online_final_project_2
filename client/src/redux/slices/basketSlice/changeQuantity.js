import { createAsyncThunk } from '@reduxjs/toolkit';
import { getToken } from '../../../localstorage/localstorage';
import findProductInBasket from './findProductInBasket';
import { AppError, handleAppError2 } from '../../../errors/errors';
import { deleteFromCart, updateCart } from '../../../api/cart';
import createRemoteCartProduct from './createRemoteCartProduct';
import basketProductCreator from './basketProductCreator';

const changeQuantityInBasket = createAsyncThunk(
  'change-quantity-in-basket',
  async (
    { product, addToBasketQuantity },
    { getState, rejectWithValue, dispatch }
  ) => {
    const idKey = '_id';
    const isLoggedIn = getToken() && true;
    const cart = JSON.parse(JSON.stringify(getState().basket.itemsBasket));
    const currentProductInBasket = findProductInBasket(product, cart);
    const productQuantityInBasket = currentProductInBasket
      ? currentProductInBasket.cartQuantity
      : 0;
    const availableQuantity = product.quantity;
    let newQuantityInBasket = productQuantityInBasket + addToBasketQuantity;
    if (newQuantityInBasket > availableQuantity) {
      return rejectWithValue({
        message: `Sorry, there are only ${availableQuantity} units of this product in stock`,
        type: 'quantityError',
      });
    }
    if (newQuantityInBasket < 0) newQuantityInBasket = 0;
    if (currentProductInBasket)
      currentProductInBasket.cartQuantity = newQuantityInBasket;
    else
      cart.push(
        basketProductCreator({ product, cartQuantity: newQuantityInBasket })
      );
    if (isLoggedIn) {
      const remoteCart = cart.map(({ product: currentProduct, cartQuantity }) =>
        createRemoteCartProduct(currentProduct, cartQuantity)
      );
      //   console.log(remoteCart);
      return handleAppError2(dispatch)(() => updateCart(remoteCart));
      //   const newCart = await updateCart(remoteCart);
      //   console.log(newCart);
      //   return newCart;
    }
    return cart;
  }
);

export const changeQuantityInBasketActionCreator = (
  product,
  addToBasketQuantity
) =>
  changeQuantityInBasket({
    ...basketProductCreator({ product, cartQuantity: 1 }),
    addToBasketQuantity,
  });

export default changeQuantityInBasket;
