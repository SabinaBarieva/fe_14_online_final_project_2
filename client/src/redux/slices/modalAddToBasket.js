import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  statusModal: false,
  modaladdimg: '',
  modaladdname: '',
  product: '',
};
const modalAddToBasketSlice = createSlice({
  name: 'modalBasket',
  initialState,
  reducers: {
    modalAddBasket: (state, { payload: currentProduct }) => {
      state.statusModal = true;
      state.modaladdimg = currentProduct.imageUrls;
      state.modaladdname = currentProduct.name;
      state.product = currentProduct;
    },
    closeModalAddBasket: (state) => {
      state.statusModal = false;
    },
  },
});

export const { modalAddBasket, closeModalAddBasket } =
  modalAddToBasketSlice.actions;
export default modalAddToBasketSlice.reducer;
