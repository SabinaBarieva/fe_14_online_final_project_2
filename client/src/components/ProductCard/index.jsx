import React, { useState } from 'react';
import { Link, useLocation } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Card, CardContent, CardMedia, Typography, Box } from '@mui/material';
import { styled } from '@mui/system';
import SvgIcon from '@mui/material/SvgIcon';
import CartIcon from '../Icons/cartIcon/cartIcon';
import { RadiusButton } from '../Buttons';
import { setProduct } from '../../redux/slices/productSlice';
import { addToBasket } from '../../redux/slices/basketSlice';

const CardContainer = styled(Card)(({ theme }) => ({
  outline: 'solid 1px transparent',
  marginBottom: '5%',
  minWidth: 'fit-content',
  backgroundColor: 'transparent',
  [theme.breakpoints.between('xs', 'md')]: {
    maxWidth: '150px',
  },
  [theme.breakpoints.between('md', 'lg')]: {
    maxWidth: '183px',
  },
  [theme.breakpoints.up('lg')]: {
    maxWidth: '203px',
  },
}));

const CardImg = styled(CardMedia)(({ theme }) => ({
  minWidth: '160px',
  display: 'block',
  minHeight: '200px',
  borderRadius: '10px',
  marginBottom: '5%',
  maxHeight: '296px',
  maxWidth: '203px',
}));

const DetailButton = styled(RadiusButton)(({ theme }) => ({
  fontSize: '0.625rem',
  lineHeight: '0.75rem',
  maxWidth: '104px',
  maxHeight: '26px',
  [theme.breakpoints.between('xs', 'md')]: {
    textTransform: 'capitalize',
    minWidth: '77px',
    minHeight: '17px',
  },
  [theme.breakpoints.between('md', 'lg')]: {
    fontSize: '0.75rem',
    minWidth: '111px',
    minHeight: '28px',
  },
  [theme.breakpoints.up('lg')]: {
    fontSize: '0.625rem',
    minWidth: '104px',
    minHeight: '26px',
  },
}));

const AddToCartBtn = styled(RadiusButton)(({ theme }) => ({
  backgroundColor: 'white',
  /*  minWidth: '29px',
  minHeight: '17px', */
  maxWidth: '40px',
  maxHeight: '26px',
  position: 'relative',
  '&:hover': {
    backgroundColor: theme.palette.primary.buttonhover,
  },
  '& svg': {
    marginTop: '5%',
    minWidth: '13px',
    minHeight: '12px',
    maxWidth: '18px',
    maxHeight: '20px',
  },
  [theme.breakpoints.between('xs', 'md')]: {
    minWidth: '29px',
    minHeight: '17px',
    '& svg': {
      width: '13px',
      height: '12px',
    },
  },
  [theme.breakpoints.between('md', 'lg')]: {
    minWidth: '42px',
    minHeight: '29px',
    '& svg': {
      width: '18px',
      height: '20px',
    },
  },
  [theme.breakpoints.up('lg')]: {
    minWidth: '40px',
    minHeight: '26px',
    '& svg': {
      width: '17px',
      height: '18px',
    },
  },
}));

const CardInfo = styled(CardContent)(({ theme }) => ({
  color: theme.palette.primary.light,
  fontFamily: theme.typography.fontFamily.secondary,
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'baseline',
  padding: '0',
  paddingLeft: '15%',
  marginTop: '3%',
  width: '80%',
  [theme.breakpoints.between('xs', 'sm')]: {
    display: 'block',
    textAlign: 'left',
    paddingLeft: '5%',
  },
  '&:last-child': {
    paddingBottom: '0',
  },
}));

const ProductName = styled(Typography)(({ theme }) => ({
  fontWeight: '700',
  [theme.breakpoints.between('xs', 'sm')]: {
    fontSize: '1rem',
    lineHeight: '1rem',
    marginBottom: '2%',
  },
  [theme.breakpoints.up('xl')]: {
    fontSize: '.875rem',
    lineHeight: '1rem',
  },
}));

const ProductPrice = styled(Typography)(({ theme }) => ({
  fontWeight: '400',
  fontSize: '.875rem',
  lineHeight: '1rem',
  /* [theme.breakpoints.between('xs', 'sm')]: {
    marginBottom: '10px',
  }, */
}));

function ProductCard({ product }) {
  const dispatch = useDispatch();
  const location = useLocation();
  const currentPath = location.pathname;
  const handleDetailClick = () => {
    dispatch(setProduct(product));
  };
  const handleAddToCartClick = () => {
    dispatch(setProduct(product));
  };
  const onClickAdd = () => {
    const item = {
      name: product.name,
      itemNo: product.itemNo,
      imageUrls: product.imageUrls,
      currentPrice: product.currentPrice,
      quantity: product.quantity,
      count: 0,
    };
    dispatch(addToBasket(item));
  };

  return (
    <CardContainer sx={{ boxShadow: 'none' }}>
      <Box sx={{ position: 'relative' }}>
        <CardImg
          component="img"
          alt={product.categories}
          image="https://hotline.ua/img/tx/343/3437076345.jpg"
        />
        <Box
          sx={{
            bottom: '2%',
            left: '50%',
            width: '100%',
            position: 'absolute',
            display: 'flex',
            justifyContent: 'center',
            transform: 'translate(-50%, -50%)',
          }}>
          <Link to={`/product/${product.itemNo}`} style={{ marginRight: '7%' }}>
            <DetailButton onClick={handleDetailClick}>Detail</DetailButton>
          </Link>
          <AddToCartBtn onClick={onClickAdd} variant="solid" disabled={false}>
            <SvgIcon
              sx={{
                position: 'absolute',
                top: '50%',
                left: '51%',
                transform: 'translate(-50%, -50%)',
              }}>
              <CartIcon />
            </SvgIcon>
          </AddToCartBtn>
        </Box>
      </Box>
      {currentPath !== '/' ? (
        <CardInfo>
          <ProductName
            variant="h2"
            sx={{
              fontSize: {
                sm: '1rem',
                lg: '0.875rem',
              },
            }}>
            {product.name}
          </ProductName>
          <ProductPrice>
            {`\u0024`}
            {product.currentPrice}
          </ProductPrice>
        </CardInfo>
      ) : null}
    </CardContainer>
  );
}

ProductCard.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string.isRequired,
    currentPrice: PropTypes.number.isRequired,
    categories: PropTypes.string.isRequired,
    imageUrls: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    quantity: PropTypes.number.isRequired,
    color: PropTypes.string.isRequired,
    brand: PropTypes.string.isRequired,
    storage: PropTypes.string.isRequired,
    itemNo: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    guarantee: PropTypes.string.isRequired,
  }).isRequired,
};

export default ProductCard;
