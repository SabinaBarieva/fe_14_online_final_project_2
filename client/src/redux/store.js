import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import productsSlice from './slices/productsSlice';
import categoriesSlice from './slices/categoriesSlice';
import errorsSlice from './slices/errorsSlice';
import modalSlice from './slices/modalSlice';
import formSlice from './slices/formSlice';
// eslint-disable-next-line import/no-named-as-default
import productSlice from './slices/productSlice';
import filtersSlice from './slices/filtersSlice';
import basketSlice from './slices/basketSlice';
import searchSlice from './slices/searchSlice';
import searchResultsSlice from './slices/searchResultsSlice';

const rootReducer = combineReducers({
  products: productsSlice,
  categories: categoriesSlice,
  errors: errorsSlice,
  modal: modalSlice,
  form: formSlice,
  product: productSlice,
  filters: filtersSlice,
  basket: basketSlice,
  search: searchSlice,
  searchList: searchResultsSlice,
});

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['basket'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});
export const persistor = persistStore(store);
export default store;
