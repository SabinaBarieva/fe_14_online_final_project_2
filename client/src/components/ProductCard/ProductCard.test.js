import React from 'react';
import {
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import ProductCard from './index';
import store from '../../redux/store';
import { ThemeProvider } from '@mui/material/styles';
import theme from '../../themes/theme';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { changeQuantityInBasketActionCreator } from '../../redux/slices/basketSlice/changeQuantity';
import { modalAddBasket } from '../../redux/slices/modalAddToBasket';
import renderer from 'react-test-renderer';

const productInStock = {
  name: 'AirPods 3rd generation with Lightning Charging Case',
  currentPrice: 181,
  sale: false,
  categories: 'headphones',
  imageUrls: [
    'headphone/egj1bm6mwsa0imaesroo.jpg',
    'headphone/kcytnfnwrjttjvxcu0zr.jpg',
    'headphone/m8lm5oq0h7bj1pv1luzx.jpg',
    'headphone/cv0ujhkjeibecrqxojzp.jpg',
  ],
  quantity: 26,
  color: 'White',
  brand: 'Apple',
  storage: '64gb',
  itemNo: '82656',
  description:
    'Apple AirPods 3 wireless headphones are headphones with high-quality sound...',
  guarantee: '3 months',
  arrivalPhoto: 'headphone/vlw1uhpqs7ofoiku9wuu.jpg',
};

const productOutOfStock = {
  name: 'AirPods 3rd generation with Lightning Charging Case',
  currentPrice: 181,
  sale: false,
  categories: 'headphones',
  imageUrls: [
    'headphone/egj1bm6mwsa0imaesroo.jpg',
    'headphone/kcytnfnwrjttjvxcu0zr.jpg',
    'headphone/m8lm5oq0h7bj1pv1luzx.jpg',
    'headphone/cv0ujhkjeibecrqxojzp.jpg',
  ],
  quantity: 0,
  color: 'White',
  brand: 'Apple',
  storage: '64gb',
  itemNo: '82656',
  description:
    'Apple AirPods 3 wireless headphones are headphones with high-quality sound...',
  guarantee: '3 months',
  arrivalPhoto: 'headphone/vlw1uhpqs7ofoiku9wuu.jpg',
};

const renderCard = (product, path) => {
  render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={[path]}>
          <ProductCard product={product} />
        </MemoryRouter>
      </ThemeProvider>
    </Provider>
  );
};
describe('ProductCard', () => {
  test('renders product card correctly with product in stock on product page', () => {
    renderCard(productInStock, '/product');
    const productCard = screen.getByTestId('product-card');
    expect(productCard).toBeInTheDocument();
    const productPrice = screen.getByText('Detail');
    expect(productPrice).toBeInTheDocument();
    const productButtons = screen.getAllByRole('button');
    expect(productButtons.length).toBeGreaterThan(0);
    const productInfo = screen.queryByTestId('product-info');
    expect(productInfo).toBeInTheDocument();
  });
  test('renders product card correctly with product out of stock', () => {
    renderCard(productOutOfStock, '/product');
    const productCard = screen.getByTestId('product-card');
    expect(productCard).toBeInTheDocument();
    const labelOutOfStock = screen.getByTestId('label-out-of-stock');
    expect(labelOutOfStock).toBeInTheDocument();
  });
  test('render product card on home page', () => {
    renderCard(productInStock, '/');
    const productCard = screen.getByTestId('product-card');
    expect(productCard).toBeInTheDocument();
    const productInfo = screen.queryByTestId('product-info');
    expect(productInfo).not.toBeInTheDocument();
  });
});

test('ProductCard snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <ProductCard product={productInStock} />
        </BrowserRouter>
      </ThemeProvider>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Product card user interactions', () => {
  test('user interaction with "Detail" and "AddToCart" buttons', async () => {
    render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <ProductCard product={productInStock} />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
    const detailButton = screen.getByTestId('detail-btn');
    fireEvent.click(detailButton);
    await waitFor(() => {
      expect(window.location.pathname).toBe('/product/82656');
    });
    const addToCart = screen.getByTestId('add-to-cart');
    fireEvent.click(addToCart);
    await waitFor(() => {
      store.dispatch(changeQuantityInBasketActionCreator(productInStock, 1));
      store.dispatch(modalAddBasket(productInStock));
    });
  });
});


