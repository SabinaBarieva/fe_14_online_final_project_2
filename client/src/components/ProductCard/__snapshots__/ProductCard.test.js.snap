// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard snapshot 1`] = `
<div
  className="MuiBox-root css-1nbic85"
  data-testid="product-card"
>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "borderRadius": "15px",
        "boxShadow": "0px 10px 20px -5px rgba(0, 0, 0, 0.4)",
        "height": "200px",
        "transform": "none",
        "width": "150px",
      }
    }
  >
    <div
      alt="AirPods 3rd generation with Lightning Charging CaseWhite"
      className="sc-bcHOTY esWDSl"
      height="200px"
      imageurl="https://res.cloudinary.com/dtvbxgclg/image/upload/headphone/vlw1uhpqs7ofoiku9wuu.jpg"
      sale="false"
      size="cover"
      width="150px"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-e553mr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FavoriteBorderIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
          />
        </svg>
      </button>
      <div
        className="MuiBox-root css-gfpxvl"
      >
        <a
          href="/product/82656"
          onClick={[Function]}
          style={
            Object {
              "marginRight": "7%",
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1bo1ka4-MuiButtonBase-root-MuiButton-root"
            data-testid="detail-btn"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Detail
          </button>
        </a>
        <div
          style={
            Object {
              "transform": " scale(1)",
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-solid MuiButton-solidPrimary MuiButton-sizeMedium MuiButton-solidSizeMedium MuiButton-root MuiButton-solid MuiButton-solidPrimary MuiButton-sizeMedium MuiButton-solidSizeMedium css-mmllqa-MuiButtonBase-root-MuiButton-root"
            data-testid="add-to-cart"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1krgx62-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <svg
                fill="none"
                height="19"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clipPath="url(#clip0_53_766)"
                >
                  <path
                    d="M7.34968 16.8999C7.74505 16.8999 8.06557 16.5646 8.06557 16.151C8.06557 15.7374 7.74505 15.4021 7.34968 15.4021C6.9543 15.4021 6.63379 15.7374 6.63379 16.151C6.63379 16.5646 6.9543 16.8999 7.34968 16.8999Z"
                    stroke="#616467"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                  />
                  <path
                    d="M15.2244 16.8999C15.6198 16.8999 15.9403 16.5646 15.9403 16.151C15.9403 15.7374 15.6198 15.4021 15.2244 15.4021C14.8291 15.4021 14.5085 15.7374 14.5085 16.151C14.5085 16.5646 14.8291 16.8999 15.2244 16.8999Z"
                    stroke="#616467"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                  />
                  <path
                    d="M1.62231 1.17261H4.48588L6.40447 11.2006C6.46993 11.5454 6.64924 11.8552 6.91099 12.0756C7.17275 12.296 7.50028 12.4131 7.83625 12.4064H14.7947C15.1307 12.4131 15.4582 12.296 15.72 12.0756C15.9817 11.8552 16.161 11.5454 16.2265 11.2006L17.3719 4.91721H5.20177"
                    stroke="#616467"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                  />
                </g>
                <defs>
                  <clipPath
                    id="clip0_53_766"
                  >
                    <rect
                      fill="white"
                      height="17.9741"
                      transform="translate(0.906494 0.423706)"
                      width="17.1814"
                    />
                  </clipPath>
                </defs>
              </svg>
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
